chore(bling): update to SHINE styling

dancormier opened the pull request:
[SPARK-58](https://stackoverflow.atlassian.net/browse/SPARK-58)

---

https://deploy-preview-2013--stacks.netlify.app/product/components/bling/

### TODO
- [x] Component styles are updated in Stacks Classic
     - [x] Add default style
     - [x] Add `filled` modifier
     - [x] Add `size` modifiers (`sm`, `lg`, default)
     - [x] Add `activity` variant
     - [x] Add `rep` variant
- [x] Add breaking changes to the v2 → v3 migration guide (see [SPARK-60](https://stackoverflow.atlassian.net/browse/SPARK-60))
- [x] Rename to `bling`
     - [x] Component uses `s-bling` class
     - [x] Documentation refers to component as `Bling`
- [x] Update documentation with new modifier/variant examples
- [x] Update documentation with copy to describe newly-updated component
- [x] Update Svelte component
     - [x] Update name to Bling
     - [x] Add `filled` bool prop
     - [x] Add `size` string prop
     - [x] Add `activity`, `rep`, null (default) to `type` prop
     - [x] Remove `children` snippet prop
     - [x] Update markup to match updated markup
     - [x] Update examples to reflect above changes
- [x] Add changeset 

### Follow up
We'll need to follow up on this PR once a few other changes have been made to the system:

- [x] The `activity` variant will need to be changed to `pink` colors once added
- [x] Visual regression images will need to be regenerated once color palettes are updated

### The docs

I have not put too much consideration into the docs. @CGuindon if you have an idea of how these should be updated, let me know and I'll add those changes.

**Update Mon Nov 10, 2025**: I've updated the docs per the Figma as [suggested](https://github.com/StackExchange/Stacks/pull/2013#pullrequestreview-3377999470).

dancormier committed:
chore(bling): update to SHINE styling

dancormier committed:
Let's stick with "award bling" for now

dancormier committed:
Why was I doing `--`?

dancormier committed:
update tests

dancormier committed:
Update other baseline images

dancormier committed:
Update svelte component

dancormier committed:
update tests, formatting

dancormier committed:
Fix some tests

dancormier committed:
got a little lint in my pocket

dancormier committed:
Change name to bling

dancormier committed:
formatting!

giamir reviewed (commented):
This component looks great @dancormier. ❤️🎉

I was lurking around and I wonder if we should add in the acceptance criteria of our stories also the **documentation of breaking changes**. I know that there is a ticket in the backlog to create a migration doc (v2 to v3). Maybe it would make sense to fill up that document as we go so that our memory is fresh on the breaking changes we made from the point of view of the consumers. 

Also, a friendly reminder to add the changesets for this PR. I think this is the first time we’ll have two changesets for two different libraries in the same PR. In any case, `npx changeset` should guide us through creating these slightly more complex changesets. 🙂

dancormier committed:
Merge branch 'beta' into SPARK-58/bling

dancormier committed:
Add TODO for migration guide

CGuindon reviewed (commented):
Docs: I have text in the figma under each style and on the left side for the "header + intro" text.
<img width="985" height="450" alt="Screenshot 2025-10-24 at 10 23 42 AM" src="https://github.com/user-attachments/assets/376ce94c-d6ad-4f67-bbb4-c2ebf2810372" />

CGuindon commented:
I guess the order of this component changes in the list as well with the new name — "Bling" should go under "Banners"
<img width="243" height="236" alt="Screenshot 2025-10-24 at 10 25 05 AM" src="https://github.com/user-attachments/assets/8705ad72-7541-4e57-b6bd-e641f73282a2" />

CGuindon commented:
All of the badge shapes should _fit_ in a 10x10 "box" but I've tweaked the sizing a bit for each to make them visually more balanced when next to each other. 

- `s-bling__bronze` should be 8x8 square (inside the 10x10 box frame) <img width="363" height="112" alt="Screenshot 2025-10-24 at 10 26 37 AM" src="https://github.com/user-attachments/assets/41906f5f-7fb2-445c-b20d-119cadf792d6" />

- `s-bling__silver`  `s-bling__gold` both look like they're the correct 9x9 size inside that box ✅
<img width="347" height="110" alt="Screenshot 2025-10-24 at 10 30 30 AM" src="https://github.com/user-attachments/assets/2abf8284-18e3-42a0-87b4-ce01bc863c06" />
<img width="323" height="102" alt="Screenshot 2025-10-24 at 10 32 37 AM" src="https://github.com/user-attachments/assets/6fde1823-f040-4acd-a190-90cf5adc2a98" />

dancormier committed:
Address PR review

dancormier committed:
Merge branch 'beta' into SPARK-58/bling

dancormier committed:
Remove award bling test images

dancormier committed:
update remaining visual regression test images

dancormier committed:
Update activity variant to use pink colors

dancormier committed:
changeset

giamir commented on packages/stacks-docs/_data/site-navigation.json:
```diff
@@ -222,6 +218,10 @@
             },
+            {
+              "title": "Bling",
+              "url": "/product/components/bling/"
```
Set `new` to true here to show the badge

giamir commented on packages/stacks-docs/product/components/bling.html:
```diff
@@ -0,0 +1,144 @@
+---
+layout: page
+title: Bling
+razor: https://razor.stackoverflow.design/components/award-bling
```
There should be a link to svelte here, not to Razor. Probably you lost some context rebasing. We don't have Razor links in the beta branch anymore.

giamir reviewed (commented):
Looks good @dancormier. I have only left a couple of minor NIT comments. 

I noticed that the following components depending on Bling are now somewhat broken:
- Badge
- Topbar
- UserCard
- Table

I guess this is on purpose and we will take care of them on follow up tickets. Correct?

Thanks for taking care of this one. ❤️

dancormier committed:
Address review

dancormier committed:
Update docs, tests to use updated badge, migration guide

dancormier committed:
whoops

dancormier commented:
> I noticed that the following components depending on Bling are now somewhat broken:
> 
>     * Badge
> 
>     * Topbar
> 
>     * UserCard
> 
>     * Table
> 
> 
> I guess this is on purpose and we will take care of them on follow up tickets. Correct?

@giamir Updates to consuming components' _styles_ should usually be left for when those components are being updated to SHINE styles, _but_ I think we should generally update any markup when there are breaking changes like there are here with bling.

I've updated the bling markup in those other components' docs pages and any relevant tests to match the expected markup. I've also updated the migration guide to mention the markup changes. If nothing else, these doc updates just make things look much nicer while we wait to update the other components 🙂 

<details>
<summary>Screenshots</summary>

<img width="165" height="143" alt="image" src="https://github.com/user-attachments/assets/91796fcd-2c78-4dcd-ab1c-3648fb137253" />
<img width="85" height="163" alt="image" src="https://github.com/user-attachments/assets/9232b1ce-4390-4e7d-9097-f87710a17bc9" />
<img width="81" height="339" alt="image" src="https://github.com/user-attachments/assets/3fce641e-7d93-4ff0-a66a-cf5b04f964c5" />
<img width="277" height="202" alt="image" src="https://github.com/user-attachments/assets/4fa1023f-cf44-45d3-960e-e15882de7218" />
<img width="263" height="78" alt="image" src="https://github.com/user-attachments/assets/d6bec3eb-3d0d-4169-afff-719eeb241771" />
<img width="196" height="76" alt="image" src="https://github.com/user-attachments/assets/49950f3d-3a7f-4f8a-ac03-a0a1d9198da3" />

</details>

> Thanks for taking care of this one. ❤️

My pleasure and thank you for the review!

giamir reviewed (approved):
Looks good Dan. Thanks for the updates. The only thing missing is the link to the svelte component (svelte badge) in the Bling doc page (in the frontmatter `svelte: ...`). Apart from that looks good to be merged from my end. 🙂

CGuindon reviewed (changes requested):
In the Svelte component the rep bling is using `black-400` — both **rep** and **default** bling dots use `black-600`. The difference is only if the fill is showing, the background color of the square box is lighter/darker but the dots are 600 in both cases. (Classic looks fine)

<img width="461" height="812" alt="Screenshot 2025-11-12 at 1 36 39 PM" src="https://github.com/user-attachments/assets/6248b6c4-2e51-4226-8c32-aed87442ad08" />

dancormier committed:
Add svelte link

dancormier committed:
Change rep bling color

dancormier committed:
Omit rep, activity base bling from svelte examples

dancormier committed:
Make the base bling appear as the base bling in svelte docs

dancormier commented:
> In the Svelte component the rep bling is using `black-400` — both **rep** and **default** bling dots use `black-600`. The difference is only if the fill is showing, the background color of the square box is lighter/darker but the dots are 600 in both cases. (Classic looks fine)

Whoops! Good catch @CGuindon. I've updated the `rep` bling to use `black-600` and I've also updated the [Svelte docs](https://deploy-preview-2013--stacks-svelte.netlify.app/?path=/docs/components-bling--docs) to omit the examples of the non-filled `rep` and `activity` variants (though a consumer could still _technically_ use them.

<details>
<summary>Screenshot</summary>

<img width="834" height="1632" alt="image" src="https://github.com/user-attachments/assets/463bd2af-e373-4314-b4d3-32cc1888b33b" />

</details>

dancormier committed:
Merge branch 'beta' into SPARK-58/bling

dancormier committed:
Update docs

mukunku commented:
Should the filled variant mention the filled class in the table below? 

<img width="910" height="561" alt="{904B1646-0917-49C3-87AA-1A92FE7EC8F3}" src="https://github.com/user-attachments/assets/8f7c485d-8e38-47b9-bb9a-8e764779b986" />

dancormier commented:
> Should the filled variant mention the filled class in the table below?
> <img alt="{904B1646-0917-49C3-87AA-1A92FE7EC8F3}" width="910" height="561" src="https://private-user-images.githubusercontent.com/4502154/514045551-8f7c485d-8e38-47b9-bb9a-8e764779b986.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjMxMzczOTIsIm5iZiI6MTc2MzEzNzA5MiwicGF0aCI6Ii80NTAyMTU0LzUxNDA0NTU1MS04ZjdjNDg1ZC04ZTM4LTQ3YjktYmI5YS04ZTc2NDc3OWI5ODYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MTExNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTExMTRUMTYxODEyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NzVlNDYxNzU1M2NkZmQxZTI3YWMxMGRmODJhMWYyZGQ3YmE2ZDQwMDI1ODU2MDg1MGFiNjYzMTE3Y2JlMjRjNSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.riRjoNxSC7OY34wYGVOiXKzSp86CV0usNuDxO1f7Z04">

@mukunku yes, nice catch! I'll add that class into the docs

dancormier committed:
Minor tweaks to the docs
